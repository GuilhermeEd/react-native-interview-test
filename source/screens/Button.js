import React, { Component } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import styles from './styles';

export default class Button extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      onPress,
      text,
    } = this.props;
  
    return (
      <TouchableOpacity onPress={onPress}><View style={styles.button}><Text>{text}</Text></View></TouchableOpacity>
    )
  }
}
