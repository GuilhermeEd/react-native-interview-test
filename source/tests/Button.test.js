import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../screens/Button';

test('<Button> renders correctly', () => {
    const onPress = () => {};
    const text = 'text';
    const tree = renderer.create(<Button onPress={onPress} text={text}/>).toJSON();
    expect(tree).toMatchSnapshot();
});