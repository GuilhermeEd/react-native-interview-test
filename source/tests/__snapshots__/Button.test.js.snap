// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#cbe6e6",
        "borderRadius": 5,
        "flex": 1,
        "justifyContent": "center",
        "margin": 10,
        "padding": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      text
    </Text>
  </View>
</View>
`;
